src/ValidationError.js

```diff
@@ -905,14 +918,23 @@ class ValidationError extends Error {
         return `${dataPath} should be equal to constant ${this.getSchemaPartText(
           error.parentSchema
         )}`;
-      case 'not':
-        return `${dataPath} should not be ${this.getSchemaPartText(
-          error.schema
-        )}${
-          likeObject(error.parentSchema)
-            ? `\n${this.getSchemaPartText(error.parentSchema)}`
-            : ''
-        }`;
+      case 'not': {
+        const postfix = likeObject(error.parentSchema)
+          ? `\n${this.getSchemaPartText(error.parentSchema)}`
+          : '';
+        const schemaOutput = this.getSchemaPartText(
+          error.schema,
+          null,
+          false,
+          false
+        );
+
+        if (likeNumber(error.schema) || likeInteger(error.schema)) {
+          return `${dataPath} should be any non-${schemaOutput}${postfix}`;
+        }
```

Comment:
Just idea - maybe we can solve all `not` logic in common way? For string/number/etc, because strings can have `minLength` and etc properties too

Reply:
I think we can, you think it is a scope of this PR?

Reply:
I think better support `not` is one PR :smile: but we can separate this on difference parts, need small roadmap

Reply:
> I think better support `not` is one PR

Agree. I will take a look =)